### ISSUE TITLE
Graphical Desktop Unable to Connect without Agent Restart

### ISSUE THREAD

**Issue by scottalanmiller** [Ylianst/MeshCentral#1174]
This is a tough one as it's being reported to me by the team. We have hundreds of users and over a thousand devices, so I'm struggling to observe the issue first hand. But we are getting regular reports of many different machines, all appear to be Windows 10 but statistically that's pretty likely, where we go and try to connect to the desktop and it SAYS that it is connected, but the screen never actually connects and the space remains black. It's not a blank screen, it never shows the screen at all.  Sometimes it starts working after a minute or two, but mostly they've taken to restarting the MeshCentral Agent process via another tool and then it connects consistently after the restart.

This has been going on for a week, maybe two.  We update automatically and are on 0.5.3 right now.  We have it happen on different networks, several different computers, different days, and through several MC version updates.

**Comment by krayon007** [comment]
Are you able to restart the agent with MC itself? On the desktop tab, click the tools button in the lower right, and find mesh background service, and click restart...

But anyways, when this happens clicking disconnect and reconnect doesn't work? Does files and terminal work when this happens? Does the console tab respond?

**Comment by scottalanmiller** [comment]
Have not tried, that, will ask for that to be tested.

**Comment by scottalanmiller** [comment]
I managed to see one of these when it happened and terminal did NOT respond.  But it was only once and came back quickly.  So I don't know if it is indicative of anything. I'll have them test that, too.

**Comment by Ylianst** [comment]
One theory is that when we release a new server with an updated meshcore, but agent is the same, the core update messes up and leaves some agents in a bad state. If this is the case, Bryan is coming out soon with fixes that will solve this.

**Comment by scottalanmiller** [comment]
Awesome, thanks. Let me know when we should start testing :)

**Comment by krayon007** [comment]
Just to clarify, what Ylian is talking about doesn't leave the agent in a "bad" state per se, it causes the agent to crash. The default installation of the agent on windows, is set to auto-restart the agent 60 seconds after a crash.

**Comment by CCWTech** [comment]
Just had this issue. Desktop just shows black screen. I left it sit for about 10 minutes to see if it would fix itself and it did not.

Terminal was working (I forgot to check console). I was able to go into tools and restart the mesh background agent and get in.

**Comment by Shootify** [comment]
yes, i have the same issue, when using Connect on endpoint and leave it there for some time, on the server side(desktop tab), it just shows black screen. it needs to be reconnect it for it to work

**Comment by CCWTech** [comment]
We're 8 months into this and we run into this very frequently. Any progress?
